File: simulator/src/main/java/com/jmodzelewski/simulator/model/Node.java
package com.jmodzelewski.simulator.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;
import java.util.Map;

import static javax.persistence.GenerationType.IDENTITY;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Node {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    Long id;
    String name;
    String loopback;

    NodeType type;

    @ElementCollection
    @CollectionTable(name = "interfaces")
    List<String> interfaces;

    @ElementCollection
    @CollectionTable(name = "routingTable")
    Map<String, String> routingTable;

    int actualX;
    int actualY;
    int previousX;
    int previousY;
}

File: simulator-ng-app/src/app/main/model/Node.ts
import {NodeType} from "./node-type.enum";

export class Node {

  id: string;
  name: string;

  interfaces: string[];
  routingTable : Map<string, string>;

  selected = false;
  type : NodeType;

  //Coordinates
  previousX = 0;
  previousY = 0;
  actualX = 0;
  actualY = 0;

  constructor() {
    this.interfaces = [];
    this.routingTable = new Map<string, string>();
  }
}
